### PR TITLE
feat: algolia insights

### PR DESCRIPTION
**Pull request by scottkelso**
# Description

### Issue

Need insights turned on for algolia searches

[Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/38213804/todos/7634025419)

### Solution

Add send event call on each strat card
Note that an event now sends for viewing a hit on any page, and when one is clicked.

image

# External Changes

Fixed a bug in ID of strategy

# Additional information

## How to test
Should have hits viewed and strategy clicked events in algolia dashboard

## TODO
Still need to replicate for journeys and videos
Could do with refactoring out and testing a useAlgoliaStrategies hook

### CODE REVIEW

File: apps/watch/src/components/StrategySections/StrategyCard/StrategyCard.spec.tsx [hunk 1]
```diff
@@ -9,6 +23,11 @@ describe('StrategyCard', () => {
     imageUrl: '',
     link: 'www.jesusfilm.org'
   }
+
+  beforeEach(() => {
+    mockHits(item)
+  })
```
Comment (GeronimoJohn) [inline apps/watch/src/components/StrategySections/StrategyCard/StrategyCard.spec.tsx:29]:
```suggestion
  beforeEach(() => {
    mockUseHits.mockReturnValue({
        hits: [item],
        sendEvent: jest.fn()
    } as unknown as HitsRenderState)
  })
```

File: apps/watch/src/components/StrategySections/StrategyCard/StrategyCard.spec.tsx [hunk 2]
```diff
@@ -1,5 +1,19 @@
...
+
+const mockUseHits = useHits as jest.MockedFunction<typeof useHits>
+
+function mockHits(item: StrategyItemProps) {
+  const sendEvent = jest.fn()
+  mockUseHits.mockReturnValue({
+    hits: [item],
+    sendEvent: sendEvent
+  } as unknown as HitsRenderState)
+}
```
Comment (GeronimoJohn) [inline apps/watch/src/components/StrategySections/StrategyCard/StrategyCard.spec.tsx:16]: can drop and do suggested below? Just the patter we've been doing

File: package.json [hunk 3]
```diff
@@ -145,6 +145,7 @@
     "react-hotkeys-hook": "^4.5.0",
     "react-i18next": "13.3.1",
     "react-instantsearch": "^7.5.0",
+    "react-instantsearch-router-nextjs": "^7.12.2",
```
Comment (GeronimoJohn) [inline package.json:148]: is this meant to be here?

Comment (scottkelso) [reply package.json:148]: Yea I think it was missing in this new branch.

**Review by jianwei1 — APPROVED** [review]

**Pull request merged**